src/EVM/SMT.hs

```diff
@@ -107,6 +112,26 @@ declareIntermediates bufs stores =
     encodeStore n expr =
        fromLazyText ("(define-const store" <> (T.pack . show $ n) <> " Storage ") <> exprToSMT expr <> ")"
 
+isAbstractStorage :: Expr Storage -> Bool
+isAbstractStorage = getAny . foldExpr go (Any False)
+  where
+    go :: Expr a -> Any
+    go AbstractStore = Any True
+    go _ = Any False
+
+-- | This function overapproximates the reads from the abstract
+-- storage. Potentially, it can return a location that has been
+-- overwritten in the abstract store. However, such reads will have
+-- been simplified away, if the source expression is simplified.
```

Comment:
I don't quite understand the comment about simplification, can you expand a bit?